### ISSUE TITLE
Include cstring to avoid 'memset not in scope' error

### ISSUE THREAD

**Issue by ml8mr** [vmware/open-vm-tools#13]
When compiling open-vm-tools-9.10.0-2476743 with gcc4.3, I see the following errors:

```
[   84s] xutils/xutils.cc: In function 'void xutils::SetFullscreenMonitorsHint(Glib::RefPtr<Gdk::Window>, std::vector<long int, std::allocator<long int> >)':
[   84s] xutils/xutils.cc:406: error: 'memset' was not declared in this scope
[   84s] xutils/xutils.cc: In function 'void xutils::RaiseWindowInternal(Glib::RefPtr<Gdk::Window>, Glib::RefPtr<Gdk::Window>, guint32)':
[   84s] xutils/xutils.cc:733: error: 'memset' was not declared in this scope
[   84s] xutils/xutils.cc: In function 'void xutils::ChangeEWMHWindowState(bool, Glib::RefPtr<Gdk::Window>, _GdkAtom*, _GdkAtom*)':
[   84s] xutils/xutils.cc:1254: error: 'memset' was not declared in this scope
```

The problem is due to the older version of gcc failing to automatically track down and include cstring. The following simple fix resolves the issue in my environment, and should not cause any problems in newer gcc environments. (I've tested against 4.3 and 4.8.)

```
Index: open-vm-tools-9.10.0-2476743/services/plugins/dndcp/xutils/xutils.cc
===================================================================
--- open-vm-tools-9.10.0-2476743.orig/services/plugins/dndcp/xutils/xutils.cc
+++ open-vm-tools-9.10.0-2476743/services/plugins/dndcp/xutils/xutils.cc
@@ -29,6 +29,7 @@
 #include <X11/Xlib.h>
 #include <X11/Xatom.h>
 #include <gdk/gdkx.h>
+#include <cstring>

 extern "C" {
 #include "vm_assert.h"
```

**Comment by oliverkurth** [comment]
Thanks for reporting this. However, we failed to reproduce this issue - can you give us more information about the environment?

**Comment by ml8mr** [comment]
I see this issue when building for SLES11 SP4. The build environment I am using includes both gcc-4.3 and gcc47, and I use CC and CXX environment variables to determine which version I'm using for the build. I can see the problem under either version.

Here are the configure options I'm using:

./configure --host=x86_64-suse-linux-gnu --build=x86_64-suse-linux-gnu --program-prefix= --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib64 --libexecdir=/usr/lib64 --localstatedir=/var --sharedstatedir=/usr/com --mandir=/usr/share/man --infodir=/usr/share/info --without-kernel-modules --without-root-privileges --without-procps --disable-deploypkg --without-ssl --with-x --disable-dependency-tracking --disable-static --without-xmlsecurity --without-xerces

Do you need anything more specific? (I can provide the entire build log...)

**Comment by oliverkurth** [comment]
Thanks, we will try again with that information.

**Comment by stanguturi** [comment]
>  #include <gdk/gdkx.h>
> +#include <cstring>
> 
>  extern "C" {

This was fixed in ebed1074323950b03d25b31c54741eaa128c3e0c Closing.
